src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -78,7 +79,7 @@ public bool OnUpdateTopology(Topology topology, bool forceUpdate = false)
             }
 
             if (state.Nodes.Count == 0)
-                throw new AllTopologyNodesDownException("There are no nodes in the topology.");
+                throw new DatabaseDoesNotExistException("There are no nodes in the topology.");
```

Comment:
The `Nodes` collection built from `Topology.Members` and `Topology.Rehabs` so to get this empty those two need to be empty.
As I understand it, this situation only possible if we in a delete process.
The server fix in the database handler is to prevent the client from getting empty topology

Reply:
cc @karmeli87

Reply:
Looks fine to me. Have you ever hit that exception?

Reply:
In my test, it hit that consistently